File: backend/src/main/java/com/smvita/computerseekho/service/ContactMessageService.java
package com.smvita.computerseekho.service;

import com.smvita.computerseekho.dto.ContactMessageDto;
import com.smvita.computerseekho.entity.ContactMessage;
import com.smvita.computerseekho.exception.ResourceNotFoundException;
import com.smvita.computerseekho.repository.ContactMessageRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional
public class ContactMessageService {

    private final ContactMessageRepository contactMessageRepository;
    private final EmailService emailService;

    // BRD: "the data will be saved in database and appropriate mail will
    // be sent to predefined mail-id". Distinct from the course-enquiry
    // confirmation flow (Day 3), which emails the enquirer, not the institute.
    @Value("${app.contact-notification-email:training@vita.com}")
    private String notificationEmail;

    public List<ContactMessageDto> findAll() {
        return contactMessageRepository.findAll().stream().map(this::toDto).toList();
    }

    public ContactMessageDto findById(Integer id) {
        return toDto(getEntityOrThrow(id));
    }

    public ContactMessageDto submit(ContactMessageDto dto) {
        ContactMessage message = new ContactMessage();
        message.setName(dto.name());
        message.setEmail(dto.email());
        message.setMessage(dto.message());
        ContactMessage saved = contactMessageRepository.save(message);

        emailService.sendSafely(
                notificationEmail,
                "New Get in Touch message from " + dto.name(),
                "Name: " + dto.name() + "\nEmail: " + dto.email() + "\n\nMessage:\n" + dto.message()
        );

        return toDto(saved);
    }

    public void markRead(Integer id) {
        ContactMessage message = getEntityOrThrow(id);
        message.setIsRead(true);
        contactMessageRepository.save(message);
    }

    private ContactMessage getEntityOrThrow(Integer id) {
        return contactMessageRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Contact Message", id));
    }

    private ContactMessageDto toDto(ContactMessage m) {
        return new ContactMessageDto(m.getMessageId(), m.getName(), m.getEmail(),
                m.getMessage(), m.getIsRead(), m.getCreatedAt());
    }
}

File: dotnet_backend/backend/Service/ContactMessageService.cs
using AutoMapper;
using ComputerSeekho.DTO;
using ComputerSeekho.Models;
using ComputerSeekho.Repository;

namespace ComputerSeekho.Service;

/// <summary>
/// BRD section 5: "the data will be saved in database and appropriate mail
/// will be sent to predefined mail-id".
///
/// Note the direction. An enquiry confirmation emails the *enquirer*; this
/// notifies the *institute*. The two are easy to confuse and do opposite
/// things.
/// </summary>
public class ContactMessageService : IContactMessageService
{
    private readonly IGenericRepository<ContactMessage> _repository;
    private readonly IMapper _mapper;
    private readonly IEmailService _emailService;
    private readonly ILogger<ContactMessageService> _logger;
    private readonly string _notificationEmail;

    public ContactMessageService(
        IGenericRepository<ContactMessage> repository,
        IMapper mapper,
        IEmailService emailService,
        IConfiguration configuration,
        ILogger<ContactMessageService> logger)
    {
        _repository = repository;
        _mapper = mapper;
        _emailService = emailService;
        _logger = logger;
        _notificationEmail = configuration["ContactNotificationEmail"] ?? "training@vita.com";
    }

    public async Task<ContactMessageDto> SubmitAsync(ContactMessageRequest request, CancellationToken ct = default)
    {
        var message = _mapper.Map<ContactMessage>(request);
        message.CreatedAt = DateTime.UtcNow;

        // Stored first, notified second — and that order is the whole point.
        //
        // The database row is the record; the email is only a notification.
        // Reversed, a mail outage would mean an enquiry from a prospective
        // student vanishes with no trace of it ever having arrived.
        var saved = await _repository.AddAsync(message, ct);

        await _emailService.SendAsync(
            _notificationEmail,
            $"New website message from {saved.Name}",
            $"""
             A message was submitted through the Get in Touch form.

             From  : {saved.Name}
             Email : {saved.Email}

             {saved.Message}

             Received: {saved.CreatedAt:dd MMM yyyy, HH:mm} UTC
             View it under Contact Messages in the admin panel.
             """,
            ct);

        return _mapper.Map<ContactMessageDto>(saved);
    }

    public async Task<IEnumerable<ContactMessageDto>> GetAllAsync(CancellationToken ct = default)
    {
        var rows = await _repository.GetAllAsync(ct);
        return _mapper.Map<IEnumerable<ContactMessageDto>>(rows.OrderByDescending(m => m.CreatedAt));
    }

    public async Task<bool> MarkReadAsync(int id, CancellationToken ct = default)
    {
        var message = await _repository.GetByIdAsync(id, ct);
        if (message is null) return false;

        message.IsRead = true;
        await _repository.UpdateAsync(message, ct);
        return true;
    }
}
